File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/driver/SauceryIOSDriver.java
package au.com.fullcirclesolutions.saucery.driver;

import io.appium.java_client.ios.IOSDriver;
import java.net.URL;
import org.openqa.selenium.remote.DesiredCapabilities;

public class SauceryIOSDriver extends IOSDriver {
    public SauceryIOSDriver(URL remoteAddress, DesiredCapabilities desiredCapabilities) {
        super(remoteAddress, desiredCapabilities);
    }

    public String GetSessionId() {
        return getSessionId().toString();
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 */

File: NUnit/Version3/Saucery3/Driver/SauceryIOSDriver.cs
using System;
using OpenQA.Selenium.Appium.iOS;
using OpenQA.Selenium.Remote;
using Saucery3.Util;

namespace Saucery3.Driver {
    public class SauceryIOSDriver : IOSDriver<IOSElement> {
        public SauceryIOSDriver(DesiredCapabilities desiredCapabilities)
            : base(new Uri(SauceryConstants.SAUCELABS_HUB), desiredCapabilities, TimeSpan.FromSeconds(400)) {
        }

        public SauceryIOSDriver(Uri remoteAddress, DesiredCapabilities desiredCapabilities)
            : base(remoteAddress, desiredCapabilities, TimeSpan.FromSeconds(400)) {
        }

        public string GetSessionId() {
            return SessionId.ToString();
        }

        //public override IOSElement ScrollTo(string text) {
        //    throw new NotImplementedException();
        //}

        //public override IOSElement ScrollToExact(string text) {
        //    throw new NotImplementedException();
        //}
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 * 
 */